check_mandatory_fields.sh

```diff
@@ -10,44 +10,41 @@
 
 set -e
 
-FIELDS=('displayName' 'description' 'tags' 'icon' 'globalMemoryLimit')
-
+# Arguments:
+# 1 - folder in which to search for meta.yaml files
+if [[ ! $1 ]]; then echo "Error: must specify path to devfiles, eg., $0 /path/to/che-devfile-registry/devfiles"; exit 1; fi
+devfilesDir=$1; devfilesDir=${devfilesDir%/} # trim trailing slash if present
 
+metaInfoFields=('displayName' 'description' 'tags' 'icon' 'globalMemoryLimit')
 
 # check that field value, given in the parameter, is not null or empty
 function check_field() {
-  if [[ $1 == "null" || $1 = "" ]];then
+  if [[ $1 == "null" || $1 = "" ]]; then
     return 1;
   fi
   return 0
 }
 
-readarray -d '' arr < <(find "$1" -name 'meta.yaml' -print0)
-
-for i in "${arr[@]}"
-do
-    id=$(yq r "$i" displayName | sed 's/^"\(.*\)"$/\1/')
-    full_id=${id}:${i}
-
+## loop through meta.yaml files
+for i in $(ls -1 "${devfilesDir}/"*"/meta.yaml" | sort); do
     echo "Checking devfile '${i}'"
+    id=$(grep "displayName:" "${i}" | sed -e "s#^displayName: ##" -e 's#"##g')
```

Comment:
Don't need yq for something this simple. Just look for lines starting with (^) {field} and a colon, then strip out quotation marks.